lib/puppet/provider/image_streamer_plan_script/synergy.rb

```diff
@@ -0,0 +1,35 @@
+################################################################################
+# (C) Copyright 2017 Hewlett Packard Enterprise Development LP
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# You may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+################################################################################
+
+require_relative '../image_streamer_resource'
+
+Puppet::Type::Image_streamer_plan_script.provide :synergy, parent: Puppet::ImageStreamerResource do
+  desc 'Provider for Image Streamer Plan Scripts using the Synergy variant of the OneView API'
+
+  confine true: login_image_streamer[:hardware_variant] == 'Synergy'
```

Comment:
Probably not really worth confining this. It is not under the namespace Synergy in the SDK and will not be calling a specific namespace or methods.